core/src/components/header/header.tsx

```diff
@@ -24,6 +36,76 @@ export class Header implements ComponentInterface {
    */
   @Prop() translucent = false;
 
+  async componentDidLoad() {
+    // Determine if the header can collapse
+    const canCollapse = (this.collapse && getIonMode(this) === 'ios') ? this.collapse : false;
+
+    const tabs = this.el.closest('ion-tabs');
+    const page = this.el.closest('ion-app,ion-page,.ion-page,page-inner');
+    const contentEl = tabs ? tabs.querySelector('ion-content') : page!.querySelector('ion-content');
+
+    if (canCollapse) {
+      await this.setupCollapsableHeader(contentEl, (tabs) ? tabs : page!);
+    }
+  }
+
+  componentDidUnload() {
+    if (this.contentScrollCallback) {
+      this.scrollEl.removeEventListener('scroll', this.contentScrollCallback);
+    }
+  }
+
+  private async setupCollapsableHeader(contentEl: any, pageEl: any) {
+    if (!contentEl) { console.error('ion-header requires a content to collapse, make sure there is an ion-content.'); }
+
+    await contentEl.componentOnReady();
+
+    this.scrollEl = await contentEl.getScrollElement();
+
+    readTask(() => {
+      const headers = pageEl.querySelectorAll('ion-header');
+      const mainHeader = Array.from(headers).find((header: any) => !header.collapse);
+
+      const mainHeaderIndex = createHeaderIndex(mainHeader);
+      const scrollHeaderIndex = createHeaderIndex(this.el);
+
+      if (!mainHeaderIndex || !scrollHeaderIndex) { return; }
+
+      setHeaderActive(mainHeaderIndex, false);
+
+      // TODO: Find a better way to do this
+      let remainingHeight = 0;
+      for (let i = 1; i <= scrollHeaderIndex.toolbars.length - 1; i++) {
+        remainingHeight += scrollHeaderIndex.toolbars[i].el.clientHeight;
+      }
+
+      /**
+       * Handle interaction between toolbar collapse and
+       * showing/hiding content in the primary ion-header
+       */
+      const toolbarIntersection = (ev: any) => { handleToolbarIntersection(ev, mainHeaderIndex, scrollHeaderIndex); };
+
+      readTask(() => {
+        const mainHeaderHeight = mainHeaderIndex.el.clientHeight;
+        const intersectionObserver = new IntersectionObserver(toolbarIntersection, { threshold: 0.25, rootMargin: `-${mainHeaderHeight}px 0px 0px 0px` });
+        intersectionObserver.observe(scrollHeaderIndex.toolbars[0].el);
+      });
+
+      /**
+       * Handle scaling of large iOS titles and
+       * showing/hiding border on last toolbar
+       * in primary header
+       */
+      this.contentScrollCallback = () => { handleContentScroll(this.scrollEl, mainHeaderIndex, scrollHeaderIndex, remainingHeight); };
+      this.scrollEl.addEventListener('scroll', this.contentScrollCallback);
+    });
+
+    writeTask(() => {
+      cloneElement('ion-title');
```

Comment:
if the cloned elements are just for presentation... i would love to avoid this, it will create another web components ion-title, that will also emit a ionStyle and so on, can get error prone.